fools/py/api_client.py

```
@@ -0,0 +1,193 @@
+    ) -> APIResponse:
+        url = f"{self.base_url}/{path.lstrip('/')}"
+        if params:
+            url = f"{url}?{urlencode(params)}"
+
+        merged_headers = {**self.default_headers, **(headers or {})}
+
+        data = None
+        if body is not None:
+            data = json.dumps(body).encode("utf-8")
+
+        request = Request(url, data=data, headers=merged_headers, method=method)
```

Ruff (0.15.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)